handler/enqueue_annotation.go

```diff
@@ -85,17 +85,17 @@ type EnqueueRequestForAnnotation[T client.Object] struct {
 	Type schema.GroupKind
 }
 
-var _ crtHandler.EventHandler = &EnqueueRequestForAnnotation[client.Object]{}
+var _ crtHandler.TypedEventHandler[client.Object, reconcile.Request] = &EnqueueRequestForAnnotation[client.Object]{}
```

Comment:
Internally `EventHandler` was already updated to be this type. Does this have to change?

Reply:
It is a bit clearer.